lib/logger.js

```diff
@@ -0,0 +1,653 @@
+    this[kBackoffMs] = constants.BASE_BACKOFF_MILLIS
+
+    let useHttps = true
+    let withCredentials = false
+    let tags = undefined
+    let timeout = constants.DEFAULT_REQUEST_TIMEOUT
+    let hostname = os.hostname()
+    let mac = undefined
+    let ip = undefined
+
+    if (has(options, 'level')) {
+      const level = options.level.toUpperCase()
+      if (!constants.LOG_LEVELS.includes(level)) {
+        const err = new Error('Invalid level')
+        err.meta = {
+          got: level
+        , expectedOneOf: constants.LOG_LEVELS
+        }
+        throw err
+      }
+      this.level = level
+    }
+
+    if (has(options, 'tags')) {
+      tags = options.tags
+      if (typeof tags === 'string') {
+        tags = tags.split(constants.TAGS_RE)
+      }
+      if (!Array.isArray(tags)) {
+        const err = new TypeError('tags should be passed as a String or an Array')
+        err.meta = {got: options.tags}
+        throw err
+      }
+      tags = tags
+        .filter((tag) => { return tag !== '' })
+        .map((tag) => { return tag.trim() })
+        .join(',')
+    }
+
+    if (has(options, 'meta')) {
+      const meta = options.meta
+      if (typeOf(meta) !== 'object') {
+        const err = new TypeError('meta needs to be an object of key-value pairs')
+        err.meta = {got: meta}
+        throw err
+      }
+      this[kMeta] = {...meta}
+    }
+
+    if (has(options, 'timeout')) {
+      if (!Number.isInteger(options.timeout)) {
+        const err = new TypeError('timeout must be an Integer')
+        err.meta = {got: options.timeout}
+        throw err
+      }
+      if (options.timeout > constants.MAX_REQUEST_TIMEOUT) {
+        throw new Error(
+          `timeout cannot be longer than ${constants.MAX_REQUEST_TIMEOUT}`
+        )
+      }
+      timeout = options.timeout
+    }
+
+    if (has(options, 'hostname')) {
+      checkStringParam(options.hostname, 'Hostname')
+      if (!constants.HOSTNAME_CHECK.test(options.hostname)) {
+        throw new Error('Invalid hostname')
+      }
+      hostname = options.hostname
+    }
+
+    if (has(options, 'mac')) {
+      checkStringParam(options.mac, 'MAC Address')
+      if (!constants.MAC_ADDR_CHECK.test(options.mac)) {
+        throw new Error('Invalid MAC Address format')
+      }
+      mac = options.mac
+    }
+
+    if (has(options, 'ip')) {
+      checkStringParam(options.ip, 'IP Address')
+      if (!isIP(options.ip)) {
+        throw new Error('Invalid IP Address format')
+      }
+      ip = options.ip
+    }
+
+    if (has(options, 'logdna_url')) {
+      console.warn('[Deprecated] logdna_url is deprecated.  Please use url instead.')
+      options.url = options.logdna_url
+    }
+    if (has(options, 'url')) {
+      checkStringParam(options.url, 'url')
+      if (!constants.PROTOCOL_RE.test(options.url)) {
+        const err = new Error('Invalid URL protocol')
+        err.meta = {
+          expected: 'http:// or https://'
```

Comment:
This might be a good opportunity to warn people who are not using secured http.

Reply:
How would you suggest they get warned?  We can't expect people to use `https` all of the time, so I'm not sure how/why we would consider that a warning.  If they're not taking the default URL, then it's buyer beware :]